Sources/Tracing/Docs.docc/Guides/InstrumentYourLibrary.md

```diff
@@ -0,0 +1,294 @@
+
+#### Handling outbound requests
+
+When a library makes an "outgoing" request or message interaction, it should invoke the  method of a configured instrument. This will invoke whichever instrument the end-user has configured and allow them to customize what metadata gets to be propagated. This can be by the following diagram:
+
+```
+                        ┌──────────────────────────────────┐
+                        │   Specific Tracer / Instrument   │
+                        └──────────────────────────────────┘            
+                                           │
+   instrument.inject(baggage: baggage, into: request, using: httpInjector)   
+                                           │                           
+                                  ┌────────▼──────┐                             
+                  ┌───────────────┤  Tracing API  ├────────────┐      ┌────┐
+                  │HTTPClient     └────────▲──────┘            │      │ N  │
+                  │                        │                   │      │ e  │
+                  │ ┌───────────────┐   ┌──▼──────────────┐    │      │ t  │
+┌───────────┐     │ |     make      │   │ add metadata to │    │      │ w  │
+│ User code |────▶│ │  HTTPRequest  │──▶│   HTTPRequest   │────┼─────▶│ o  │
+└───────────┘     │ └───────────────┘   └─────────────────┘    │      │ r  │
+                  │                                            │      │ k  │
+                  └────────────────────────────────────────────┘      └────┘
+```
+
+> Note: A library _itself_ cannot really know what information to propagate, since that depends on the used tracing or instrumentation system. The library does however understand its carrier type, and thus can implement the ``Injector`` protocol.
+
+For example, an HTTP client e.g. should inject the current baggage (which could be carrying trace ``Span`` information) into the HTTP headers as follows:
+
+```swift
+func get(url: String) -> HTTPResponse {
+  var request = HTTPRequest(url: url)
+  
+  if let baggage = Baggage.current {
+    InstrumentationSystem.instrument.inject(
+      baggage,
+      into: &request,
+      using: HTTPRequestInjector()
+    )
+  }
+  
+  try await _send(request)
+}
+```
+
+As you can see, the library does not know anything about what tracing system or instrumentation is installed, because it cannot know that ahead of time. 
+
+All it has to do is query for the current [task-local](https://developer.apple.com/documentation/swift/tasklocal) `Baggage` value, and if one is present, call on the instrumentation system to inject it into the request.
+
+Since neither, the tracing API, or the specific tracer backend are aware of this library's specific `HTTPRequest` type, we also need to implement an ``Injector`` which takes on the responsibility of adding the metadata into the carrier type (which in our case is the `HTTPRequest`). An injector could for example be implemented like this:
```

Comment:
```suggestion
Since neither the tracing API nor the specific tracer backend is aware of this library's specific `HTTPRequest` type, we also need to implement an ``Injector`` which takes on the responsibility of adding the metadata into the carrier type (which in our case is the `HTTPRequest`). An injector could for example be implemented like this:
```